packages/runtimes/hmr/src/loaders/hmr-runtime.js

```diff
@@ -40,7 +40,12 @@ var parent = module.bundle.parent;
 if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
   var hostname = getHostname();
   var port = getPort();
-  var protocol = location.protocol === 'https:' ? 'wss' : 'ws';
+  var protocol =
+    HMR_SECURE ||
+    (location.protocol == 'https:' &&
+      !/localhost|127.0.0.1|0.0.0.0/.test(hostname))
+      ? 'wss'
+      : 'ws';
```

Comment:
Why is this change needed?

Reply:
It was necessary to make content scripts HMR work. Effectively, on an HTTPS site, say `https://google.com` the runtime previously tried to access `wss://google.com:1234` by default. If you set the host to `localhost` from the command line, you'd still get `wss://localhost:1234`, which didn't work. Now, the system will only access `wss` if either `HMR_SECURE` was set from the server, or we're not on a local network. One would assume that on a local network, proxies that add HTTPS support are rare. Therefore, this change does not break any environments but adds support for accessing the correct URL with content scripts.